src/v/cluster/utils/notifications.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "cluster/notification.h"
+#include "cluster/topic_configuration.h"
+#include "model/fundamental.h"
+
+#include <seastar/util/noncopyable_function.hh>
+
+namespace cluster {
+
+// Unified interface for partition level notifications.
+// Supports notifications for partition leadership changes,
+// partition management (creation, deletion), and partition properties changes.
+class notifications {
```

Comment:
Will think about naming, perhaps a bit too generic. Maybe `partition_leadership_notifier`?

Reply:
how about partition_change_notifier? (since this is not limited to leadership alone)

Reply:
yeah SGTM 👍